copy.c

```diff
@@ -1,11 +1,46 @@
 #include <stdio.h>
 #include <windows.h>
+#include <fcntl.h>
+
+typedef struct node_tag {
+	char buffer[1024];
+	int capacity;
+	struct node_tag *next;
+} node;
+
+int wmain() {
+	_setmode(_fileno(stdin), _O_U8TEXT);
+
+	node* bufferList = malloc(sizeof(node));
+	bufferList->capacity = 0;
+	node* head = bufferList;
+
+	DWORD cbSize = 0;
+	DWORD count;
+	while ((count = fread(head->buffer, 1, sizeof(head->buffer), stdin)) != 0 && !ferror(stdin)) {
```

Comment:
No need for the `ferror()` check in the while loop - `fread` returns 0 for both EOF as well as error.